diff --git a/autogen.sh b/autogen.sh
index 4f5e612..4074685 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -4,5 +4,3 @@ aclocal || exit 1
 autoheader || exit 1
 autoconf || exit 1
 automake -a -c || exit 1
-./configure --enable-maintainer-mode --enable-debug-log \
-	--enable-examples-build $*
diff --git a/libusb/os/linux_usbfs.c b/libusb/os/linux_usbfs.c
index 0b746a2..956519f 100644
--- a/libusb/os/linux_usbfs.c
+++ b/libusb/os/linux_usbfs.c
@@ -123,7 +123,7 @@ static int initialize_device(struct libusb_device *dev, uint8_t busnum,
 
 	snprintf(path, PATH_MAX, "%s/%03d/%03d", usbfs_path, busnum, devaddr);
 	usbi_dbg("%s", path);
-	fd = open(path, O_RDWR);
+	fd = open(path, O_RDONLY);
 	if (!fd) {
 		usbi_dbg("open '%s' failed, ret=%d errno=%d", path, fd, errno);
 		/* FIXME this might not be an error if the file has gone away due
